### DISCUSSION TITLE
Getting the text from a full sentence

### DISCUSSION THREAD

**Discussion by prajinkhadka** [mindee/doctr#1018] [Q&A]
Is there a way to input the co-ordinate too  to doctr to extract text from that specific part. Co-ordinates, I mean the bounding box co-ordinate. 
Another questions is, 
I recently found that doctr is not optimized for cropped images. We need to pass the localized part directly to text recognition engine, that works fine for recognizing a word. But what if its a cropped image, but a full sentence. I am not able to figure out how doctr works in that case. Here is a sample image. I tried in the hugging face demo. Its not able to detect anything. Am I missing something ? 
It works fine in case Tesseract. I understand I would do word segmentation and recognize each word but is there a way to input this whole image and get the results at once. 

![2](https://user-images.githubusercontent.com/34759925/183883353-a88205da-4265-465d-bfe2-25cf0ce5a5d5.png)

**Comment by prajinkhadka** [comment]
I understand and tried,  by using pytesseract to  detect the words in the word crop and after this pass the word crops to the recognition predictor. It works as expected. 
But, to make it  work without tesseract, I reckon I need to train in my own dataset, yeah ?

**Comment by frgfm** [comment]
Hello @prajinkhadka :wave:

Here is my answer:
> Is there a way to input the co-ordinate too to doctr to extract text from that specific part. Co-ordinates, I mean the bounding box co-ordinate.

what you first describe above, is text recognition. So yes this is possible, use the coordinates to crop your image and send it to the `recognition_predictor` :) 

> But what if its a cropped image, but a full sentence

each project is free to define their own tasks, and you're free to consider a vocab where you add spaces to train your own model (possible inside docTR), but by default we discard white spaces as vocab. So our text recognition task is constrained to uninterrupted sequences of characters.


Now about your image, two options without any code changes:
- split it into word images (histogram methods should be quite sufficient) and forward it to the recognition predictor
- or feed the whole thing to the full OCR predictor (for that, considering our the model was trained, I suggest padding the image to reduce the relative size of the text)

I hope this helps!
Cheers :v:
